src/c-writer.cc

```diff
@@ -1745,6 +1823,57 @@ void CWriter::WriteImports() {
   }
 }
 
+void CWriter::WriteTailCallWeakImports() {
+  for (const Import* import : unique_imports_) {
+    if (import->kind() != ExternalKind::Func) {
+      continue;
+    }
+    const Func& func = cast<FuncImport>(import)->func;
+    Write(Newline(), "/* handler for missing tail-call on import: '",
+          SanitizeForComment(import->module_name), "' '",
+          SanitizeForComment(import->field_name), "' */", Newline());
+    Write("WEAK_FUNC_DECL(",
```

Comment:
We don't need to fix this now: but in the future we may need to have a macro that controls whether the weak func's are a part of compiled output of wasm2c. This will allow use in toolchains that have bad weak symbol support.

```c
# ifndef WASM_NO_TAILCALL_DEFAULTS
WEAK_FUNC_DECL(...);
# endif
```

Reply:
Fair enough. These weak symbols are only generated when tail calls are explicitly enabled (with `--enable-tail-call`), so I don't think there's an immediate issue. I do imagine this may become the default soon now that tail calls are at phase 4, so the story for these toolchains might turn into "you'll want to run wasm2c with `--disable-tail-call`".